File: backend/src/main/kotlin/org/team14/webty/search/dto/SearchSuggestionDto.kt
package org.team14.webty.search.dto

// 검색어 자동완성 응답 DTO 클래스입니다.
data class SearchSuggestionDto(
    // 자동완성 목록을 저장합니다.
    val suggestions: List<String>,
    // 검색 URL을 저장합니다(정렬 방식 정보 포함).
    val searchUrl: String = "/search" // 기본 검색 URL
)

File: frontend/src/lib/types/search/SearchSuggestionDto.ts
/**
 * 검색어 자동완성 응답 DTO 인터페이스
 * 백엔드 DTO와 일치하도록 업데이트됨
 */
export interface SearchSuggestionDto {
  // 자동완성 목록
  suggestions: string[];
  // 검색 URL(정렬 방식 정보 포함)
  searchUrl: string;
}

/**
 * 자동완성 항목 하나에 대한 상세 정보를 담는 DTO 인터페이스
 */
export interface SuggestionItemDto {
  // 자동완성 텍스트
  text: string;
  // 자동완성 타입("webtoonName", "nickname" 등)
  type?: string;
  // 검색 URL(정렬 방식 포함 가능)
  searchUrl: string;
} 